src/utils/image.js

```diff
@@ -408,13 +408,14 @@ export class RawImage {
             // Draw image to context, padding in the process
             ctx.drawImage(canvas,
                 0, 0, this.width, this.height,
-                left, top, newWidth, newHeight
+                left, top, this.width, this.height
```

Comment:
The first row is effectively saying "start a selection at `0,0` and `this.width, this.height`", which gives us a rectangle.

It then draws this selection on line 2, starting at a new `x, y` of `left, top`, and the new dimensions should match what the original image looks like otherwise it gets stretched.

---

You can see why the cropping happens, as shown in my first post, by just looking at width.  
We are stretching the image to `newWidth` and the new canvas is already at a width of `newWidth` hence when we add the padding of `left` to it, we now get a new image that is larger than a canvas - `left + newWidth + right` is greater than `newWidth`!

Reply:
Amazing explanation - thanks!